Revised from RT#7502.

Try to avoid writing krbLastAdminUnlock when we're just doing auditing
in the KDC.  Because the attributes that we want to update are not in
the tl_data sequence, and krbLastAdminUnlock is, we NULL it out
temporarily to avoid updating any of the attributes that are used for
holding tl_data values.

--- src/plugins/kdb/ldap/libkdb_ldap/lockout.c
+++ src/plugins/kdb/ldap/libkdb_ldap/lockout.c
@@ -149,6 +149,8 @@ krb5_ldap_lockout_audit(krb5_context con
     krb5_deltat failcnt_interval = 0;
     krb5_deltat lockout_duration = 0;
     krb5_timestamp unlock_time;
+    krb5_tl_data *tl_data;
+    krb5_int16 n_tl_data;
 
     SETUP_CONTEXT();
 
@@ -217,7 +219,13 @@ krb5_ldap_lockout_audit(krb5_context con
     }
 
     if (entry->mask) {
+        tl_data = entry->tl_data;
+        n_tl_data = entry->n_tl_data;
+        entry->tl_data = NULL;
+        entry->n_tl_data = 0;
         code = krb5_ldap_put_principal(context, entry, NULL);
+        entry->tl_data = tl_data;
+        entry->n_tl_data = n_tl_data;
         if (code != 0)
             return code;
     }
